### PR TITLE
Add covar operators

### LINKED ISSUE #1548: Add covar operators to DataFusion
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Add covar operators to data fustion.

**Describe the solution you'd like**

Add three operators: `covar`, `covar_samp` and `covar_pop`.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by realno**
# Which issue does this PR close?

Closes #1548 .

 # Rationale for this change
This PR adds `covar`, `covar_pop` and `covar_samp` statistical functions as aggregators to DataFusion.

# What changes are included in this PR?

# Are there any user-facing changes?

The following operators are added:
`covar`
`covar_pop`
`covar_samp`

No breaking changes.

### CODE REVIEW

File: datafusion/src/physical_plan/expressions/variance.rs [hunk 1]
```diff
@@ -605,6 +605,62 @@ mod tests {
         Ok(())
     }
 
+    #[test]
```
Comment (realno) [inline datafusion/src/physical_plan/expressions/variance.rs:608]: Added some test for `variance` to test merge rules.

File: datafusion/src/physical_plan/expressions/stddev.rs [hunk 2]
```diff
@@ -411,6 +411,62 @@ mod tests {
         Ok(())
     }
 
+    #[test]
```
Comment (realno) [inline datafusion/src/physical_plan/expressions/stddev.rs:414]: Added tests for merge rule.

**Comment by liukun4515** [comment]
Thank you for your works! @realno

**Review by houqp — APPROVED** [review]
LGTM

**Review by liukun4515 — APPROVED** [review]
LGTM

**Pull request merged**